src/main/java/org/zalando/nakadi/repository/db/StorageDbRepository.java

```diff
@@ -46,11 +62,29 @@ public Storage createStorage(final Storage storage) throws DataAccessException {
         } catch (final JsonProcessingException ex) {
             throw new IllegalArgumentException("Storage configuration " + storage.getConfiguration(Object.class) +
                     " can't be mapped to json", ex);
+        } catch (DuplicateKeyException e) {
```

Comment:
It looks quite handy to use this kind of exceptions here, but it is not good to use presentation layer entities from storage layer. I suppose it is better to propagate runtime exception of jdbc template and process them on the service layer.